### PR TITLE
Option to give an existing vocab file(.pt or text) to preprocess.py

### PR DESCRIPTION
**Pull request by lordeddard**
Addresses the issue [#1304](https://github.com/OpenNMT/OpenNMT-py/issues/1304). Tried to implement the first feature, ability to reuse the existing vocabulary.

### CODE REVIEW

**Comment by vince62s** [comment]
Sorry, I was not clear enough in what we wanted.
The logic is good but must be placed here: https://github.com/OpenNMT/OpenNMT-py/blob/master/onmt/inputters/inputter.py#L339
rather than in preprocess.
can you adjust ?

**Comment by lordeddard** [comment]
Yes! Will do it!

File: onmt/opts.py [hunk 1]
```diff
@@ -219,6 +219,8 @@ def preprocess_opts(parser):
     # Dictionary options, for text corpus
 
     group = parser.add_argument_group('Vocab')
+    group.add('--use_existing_vocab', '-use_existing_vocab',
+              action='store_true', default="", help="Use existing vocabulary.")
     group.add('--src_vocab', '-src_vocab', default="",
```
Comment (vince62s) [inline onmt/opts.py]: you can remove the option, it is not necessary.

File: preprocess.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def build_save_vocab(train_dataset, fields, opt):
         train_dataset, fields, opt.data_type, opt.share_vocab,
         opt.src_vocab, opt.src_vocab_size, opt.src_words_min_frequency,
         opt.tgt_vocab, opt.tgt_vocab_size, opt.tgt_words_min_frequency,
-        vocab_size_multiple=opt.vocab_size_multiple
+        opt.use_existing_vocab, vocab_size_multiple=opt.vocab_size_multiple
     )
```
Comment (vince62s) [inline preprocess.py]: remove

File: onmt/inputters/inputter.py [hunk 3]
```diff
@@ -311,7 +311,7 @@ def _build_fv_from_multifield(multifield, counters, build_fv_args,
 def build_vocab(train_dataset_files, fields, data_type, share_vocab,
                 src_vocab_path, src_vocab_size, src_words_min_frequency,
                 tgt_vocab_path, tgt_vocab_size, tgt_words_min_frequency,
-                vocab_size_multiple=1):
+                use_existing_vocab, vocab_size_multiple=1):
```
Comment (vince62s) [inline onmt/inputters/inputter.py]: remove

File: onmt/inputters/inputter.py [hunk 4]
```diff
@@ -336,6 +336,17 @@ def build_vocab(train_dataset_files, fields, data_type, share_vocab,
 
     counters = defaultdict(Counter)
 
+    if use_existing_vocab:
```
Comment (vince62s) [inline onmt/inputters/inputter.py]: you cna remove this test.

**Comment by vince62s** [comment]
looks good, thanks !

**Comment by guillaumekln** [comment]
What about the target vocabulary?

**Comment by guillaumekln** [comment]
Oh ok, they are bundled in a single .pt file.

**Comment by vince62s** [comment]
Well the vocab .pt file contains both src vocab and tgt vocab.
It would be cumbersome to force users to specify -src_vocab file.pt and -tgt_vocab file.pt
So we need to document this.

**Comment by vince62s** [comment]
@lordeddard can you please add a comment in the opt.py to be very specific ?
thanks.

**Comment by lordeddard** [comment]
Yes! Will do!

**Pull request merged**